keras_nlp/models/bert/bert_models.py

```diff
@@ -259,215 +215,83 @@ def get_config(self):
                 "max_sequence_length": self.max_sequence_length,
                 "num_segments": self.num_segments,
                 "dropout": self.dropout,
-                "cls_token_index": self.cls_token_index,
             }
         )
         return config
 
+    # TODO(jbischof): consider exposing `presets` as member variable.
+    # Need to protect dict from mutation.
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        load_weights=True,
+        name=None,
+        trainable=True,
+    ):
 
-MODEL_DOCSTRING = """Bert "{type}" architecture.
+        if preset not in presets:
+            raise ValueError(
+                "`preset` must be one of "
+                f"""{", ".join(presets)}. Received: {preset}."""
+            )
+        metadata = presets[preset]
+        config = metadata["config"]
+        config["name"] = name
+        config["trainable"] = trainable
+        model = cls(**config)
 
-    This network implements a bi-directional Transformer-based encoder as
-    described in ["BERT: Pre-training of Deep Bidirectional Transformers for
-    Language Understanding"](https://arxiv.org/abs/1810.04805). It includes the
-    embedding lookups and transformer layers, but not the masked language model
-    or classification task networks.
+        if not load_weights:
+            return model
+
+        weights = keras.utils.get_file(
+            "model.h5",
+            metadata["weights_url"],
+            cache_subdir=os.path.join("models", preset),
+            file_hash=metadata["weights_hash"],
+        )
+
+        model.load_weights(weights)
+        return model
+
+
+FROM_PRESET_DOCSTRING = """Instantiate BERT model from preset architecture and weights.
 
     Args:
-        weights: string, optional. Name of pretrained model to load weights.
-            Should be one of {names}.
-            If None, model is randomly initialized. Either `weights` or
-            `vocabulary_size` must be specified, but not both.
-        vocabulary_size: Int, optional. The size of the token vocabulary. Either
-            `weights` or `vocabulary_size` must be specified, but not both.
+        preset: string. Must be one of {names}.
+        load_weights: Whether to load pre-trained weights into model. Defaults
+            to `True`.
         name: string, optional. Name of the model.
         trainable: boolean, optional. If the model's variables should be
             trainable.
 
-    Examples:
+    # Examples:
     ```python
-    # Randomly initialized Bert{type} encoder
-    model = keras_nlp.models.Bert{type}(vocabulary_size=10000)
+    # Load architecture and weights from preset
+    model = Bert.from_preset("bert_base_uncased_en")
 
-    # Call encoder on the inputs.
+    # Call encoder on the inputs
```

Comment:
Maybe move the input data setup towards the top of this example, and keep the load/call with preset weights vs load/call random weight right next to each other. I would find that a little more readable.

Reply:
Happy to do that, but this pattern is now all over the `models/` folder so we will have some cleanup to reconcile.